treescope/_internal/api/array_autovisualizer.py

```diff
@@ -97,12 +97,17 @@ def _autovisualize_array(
     # Assign axes, using preferred axes if possible.
     row_axes = []
     column_axes = []
-    for info in array_axis_info:
-      if isinstance(info, NamedPositionalAxisInfo | NamedPositionlessAxisInfo):
-        if info.axis_name in self.prefers_column:
-          column_axes.append(info.axis_name)
-        elif info.axis_name in self.prefers_row:
-          row_axes.append(info.axis_name)
+    for column_key in self.prefers_column:
+      for info in array_axis_info:
+        if isinstance(info, NamedPositionalAxisInfo | NamedPositionlessAxisInfo | PositionalAxisInfo):
+          if info.logical_key() == column_key:
+            column_axes.append(info)
+
+    for row_key in self.prefers_row:
+      for info in array_axis_info:
+        if isinstance(info, NamedPositionalAxisInfo | NamedPositionlessAxisInfo | PositionalAxisInfo):
+          if info.logical_key() == row_key:
+            row_axes.append(info)
```

Comment:
Why change this to a double loop? I think the original loop structure will be faster and avoid unnecessary checks for axes that are not named.

Also, are you trying to add support for preferring integer axis positions? I don't think this works correctly since `NamedPositionalAxisInfo` has both a name and a position and the `logical_key` just returns one of them.